src/book/dto/create.ts

```diff
@@ -32,5 +32,5 @@ export class CreateBookDto implements Book {
   readonly publishingDate: Date;
 
   @IsNotEmpty()
-  readonly category: string;
+  readonly category: Category;
```

Comment:
Opa, aqui acho que tem um ponto legal. A sua classe livro ela é rica, representa o seu domínio e guarda o estado da aplicação. Já o seu dto, é uma classe mais pobrinha.. Ela existe para transitar os parâmetros da request para o domínio. Eu geralmente recomendo que os atributos do DTO sejam de tipos mais simples... E quando o atributo representa um relacionamento, eu sugiro receber o id. 

No caso da categoria, seria um número e o nome do atributo seria categoriaId por exemplo. E aí você precisa ter o trabalho de carregar a categoria em função do id. 

Agora uma curiosidade.. Como que o Nest monta esse objeto com essa categoria aí.. como foi json que você passou na request?

Reply:
Ahh entendi!! E concordo sobre o dto  ser `categoryId: number`, acho que ajuda também na hora de descobrir quais dados passar.

Eu to passando um objeto assim na request (ta grandão pq eu quis testar um sumário bem longo já que tinha que ser tamanho livre)

```
{
	"title":"Era umasada asda vez umsd dfsdfsdfsdf sdf...",
	"abstract":"Assumenda odio obcaecati tenetur veniam ipsa voluptates, sunt, reprehenderit, veritatis aliquid ab eligendi delectus deserunt eum recusandae rem nam nemo accusantium similique. Laborum quae, tempore rerum in cupiditate totam inventore?",
	"summary":" # Livro\n\nLorem _ipsum_ dolor sit amet consectetur adipisicing elit. Exercitationem dolor placeat, accusamus deserunt consectetur sequi accusantium quasi enim cum maiores cupiditate, laboriosam nihil totam eos facilis obcaecati officia a explicabo. Dolor facilis consectetur laborum labore laboriosam porro consequuntur praesentium veritatis maiores. Eaque aliquam incidunt impedit, **autem veritatis** animi nobis vero nostrum minima officiis velit corporis iure molestiae, maiores vel doloribus. Laboriosam impedit reprehenderit voluptas culpa aspernatur unde facere animi eaque magni, suscipit totam inventore vel, illum voluptates enim recusandae nam consequatur dolore? Error delectus ad voluptatibus facere consequuntur, facilis perspiciatis? Repellat, reprehenderit praesentium cumque rem alias, quisquam molestias cupiditate officia doloremque obcaecati mollitia consequatur repellendus fuga. Fugiat atque quo, aspernatur, cumque et, repellat minima voluptate molestias modi ipsa assumenda maxime.\n\n## Laudantium, fugiat. Obcaecati quasi dolorem sequi. Enim accusantium consequuntur corporis quas voluptates suscipit architecto ducimus repellendus cumque explicabo sequi quo necessitatibus quibusdam, dolor nihil, natus quod atque labore tempore! Voluptatibus. Doloribus cumque officia consectetur quisquam neque ducimus quam quae vitae exercitationem minus in aliquid enim a fuga doloremque repellat quo voluptatem repellendus corrupti voluptas vero, quis nostrum expedita. Earum, nulla.\n\nHic quisquam quo accusamus omnis commodi beatae eius nesciunt natus distinctio necessitatibus et corrupti iusto laboriosam minima est, similique, illo minus numquam? Laborum porro iusto sequi hic quibusdam exercitationem ut!\n\n# Sapiente eaque dolores animi voluptates, ex minima facilis! A necessitatibus nulla enim dolorum placeat sit nesciunt eum molestiae cupiditate maxime, voluptatibus impedit autem, facilis quas fugiat recusandae temporibus fuga amet.\nCommodi neque facilis excepturi deserunt sunt quae accusantium, voluptates libero blanditiis veritatis rem exercitationem asperiores accusamus eligendi sed, suscipit fugiat est. Facere delectus a nemo? Quis, voluptate. Esse, rerum minus.\nHic id, eaque iste iusto tempore eveniet autem pariatur odit voluptate animi nulla qui et inventore? Delectus saepe molestias aliquam? Repellat optio eveniet necessitatibus nam molestiae iste voluptatum repudiandae nulla?\nFugit provident nisi aperiam. Ab recusandae autem delectus. Molestias autem placeat rem! Perferendis iure molestiae molestias ratione cumque dolores nobis exercitationem rem impedit totam quae, sequi quasi, fuga nesciunt quas!\n\n### Beatae cum autem culpa, iste, dolores architecto sed ullam ab enim perferendis impedit minus at recusandae velit illum in sit nisi dolorum facere nihil corporis cupiditate laborum incidunt! Minus, nostrum.\n\n* Tempora beatae aspernatur corporis, facere quisquam, reprehenderit consectetur modi doloremque harum neque sed facilis praesentium incidunt illum! Illum maxime dignissimos beatae, at, sed ipsa fuga tempore tempora eos corrupti reiciendis.\n\n* Possimus est nam exercitationem molestiae sint aut quasi earum reiciendis quam illo, aspernatur magnam dolorem, ad quis vitae veniam dolores vero consequatur libero ipsum saepe provident eum deserunt. Cumque, cupiditate?\n",
	"price":59,
	"numberOfPages":110,
	"ISBN":1234528490125,
	"publishingDate":"2020-03-21T03:47:32.279Z",
	"category": {
		"id": 1
	}
}
```

Reply:
Legal você me mostrar o JSON. Fica até mais fácil de melhora a explicação... O mal de você receber o parâmetro da Category assim é que vocÊ monta uma category num estado inconsistente... Digo, o Nest cria uma instância do tipo Category, mas tudo está nulo menos o id. Já recebendo só o id, você consegue construir o objeto em um estado válido.